### PR TITLE
MONIT-16056 Proxy support for events (draft)

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/listeners/WavefrontPortUnificationHandler.java [hunk 1]
```diff
@@ -114,6 +120,23 @@ protected void processLine(final ChannelHandlerContext ctx, String message) {
...
         return;
+      case EVENT:
+        ReportableEntityHandler<Event> eventHandler = eventHandlerSupplier.get();
+        if (eventHandler == null || eventDecoder == null) {
+          wavefrontHandler.reject(message, "Port is not configured to accept event data!");
+          return;
+        }
+        List<Event> events = Lists.newArrayListWithCapacity(1);
+        try {
+          eventDecoder.decode(message, events, "dummy");
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/listeners/WavefrontPortUnificationHandler.java]: where is the event format defined?

File: proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java [hunk 2]
```diff
@@ -36,8 +36,10 @@
   private final AtomicInteger pointsPerBatch;
   private final AtomicInteger memoryBufferLimit;
 
-  private static final RecyclableRateLimiter sourceTagRateLimiter =
+  private static final RecyclableRateLimiter SOURCE_TAG_RATE_LIMITER =
       RecyclableRateLimiter.create(5, 10);
+  private static final RecyclableRateLimiter EVENT_RATE_LIMITER =
+      RecyclableRateLimiter.create(0.2, 10);
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java]: is this adjustable based on the backend setting?

**Comment by jbau** [comment]
we probably want to have a failsafe setting for the max disk queue

File: proxy/src/main/java/com/wavefront/agent/handlers/ReportableEntityHandlerFactoryImpl.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
   private static final Logger logger = Logger.getLogger(
       ReportableEntityHandlerFactoryImpl.class.getCanonicalName());
 
-  private static final int SOURCE_TAGS_NUM_THREADS = 2;
+  private static final int API_NUM_THREADS = 2;
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/handlers/ReportableEntityHandlerFactoryImpl.java]: possible keep SOURCE_TAG_NUM_THREADS and add EVENT_NUM_THREADS

File: proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java [hunk 4]
```diff
@@ -37,8 +37,10 @@
   private final AtomicInteger pointsPerBatch;
   private final AtomicInteger memoryBufferLimit;
 
-  private static final RecyclableRateLimiter sourceTagRateLimiter =
+  private static final RecyclableRateLimiter SOURCE_TAG_RATE_LIMITER =
       RecyclableRateLimiterImpl.create(5, 10);
+  private static final RecyclableRateLimiter EVENT_RATE_LIMITER =
+      RecyclableRateLimiterImpl.create(0.2, 10);
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java]: adjust.

File: proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java [hunk 5]
```diff
@@ -823,6 +856,41 @@ public Response removeTag(String id, String tagValue, boolean forceToQueue) {
...
+    private final List<Event> events;
+
+    public PostEventResultTask(List<Event> events) {
+      this.events = events;
+    }
+
+    @Override
+    public List<PostEventResultTask> splitTask() {
+      // currently this is a no-op
+      return ImmutableList.of(this);
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java]: don't we need splitting?

Comment (basilisk487) [reply proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java]: yup. fixed

File: proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java (hunk 4 above)
Comment (basilisk487) [reply proxy/src/main/java/com/wavefront/agent/handlers/SenderTaskFactoryImpl.java]: fixed

File: proxy/src/main/java/com/wavefront/agent/handlers/ReportableEntityHandlerFactoryImpl.java (hunk 3 above)
Comment (basilisk487) [reply proxy/src/main/java/com/wavefront/agent/handlers/ReportableEntityHandlerFactoryImpl.java]: done

**Comment by jbau** [comment]
lgtm

**Pull request merged**